lib/Runtime/Library/JavascriptString.h

```diff
@@ -40,8 +40,10 @@ namespace Js
         JavascriptString(JavascriptString&) = delete;
 
     private:
-        Field(const char16*) m_pszValue;         // Flattened, '\0' terminated contents
+        Field(const char16*) m_pszValue;          // Flattened, '\0' terminated contents
         Field(charcount_t) m_charLength;          // Length in characters, not including '\0'.
+        Field(codepoint_t*) m_codePointString;    // Code points of the string, may be nullptr if not initialized yet by GetCodePoints call
+        Field(charcount_t) m_codePointsLength;    // Length of the codepoints, may be k_InvalidCharCount if not initialized yet by GetCodePoints call
```

Comment:
This will add two pointers to every single JS string - which isn't ideal - could we instead subclass JavascriptString with a new type like JavascriptStringWithCodePoints or something?

(Note I think this change is causing the compile failures as there is a static assert based on the size of a different subclass of JavascriptString)

Reply:
So I will need to create a new instance of JavascriptStringWithCodePoints on each SimpleMatch call if pattern shows us that it's unicode one.
Honestly I cannot imagine how i will cache the codepoint strings.

Reply:
You're right this is not an ideal solution - ugh - I didn't think this through did I. I think cache'ing is a no go.

Either a) we massively increase the size of all our strings OR b) we come up with a way to convert the type whenever we need to cache - which would not be simple as there could be many existing pointers to the string.